pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -118,10 +118,35 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 				pc.deleteIP(ip)
 			}
 		}
+		// fire instance handles for workload
+		for _, handler := range pc.c.workloadHandlers {
```

Comment:
Are we sure this wont fire a billion times per second for every update to a pod like volumes mounted, secrets mojnted, etc etc. ? Should we fire this handler only when there are meaningful changes like new pod, delete pod? Essentially, if you fire this event only when there has been a call to pc.update/delete, then the number of calls will be reduced right?

Reply:
We do have filtering downstream, in redundantEventForPod based on `ForeignSeviceInstancesEqual`, so I don't think it will be horrible. Probably need to compare this to a method of some additional up front filtering. We pretty much just need the pod IP so maybe we can use that to optimize?